### PR TITLE
Adds PR template

### PR DESCRIPTION
**Pull request by ale8k**

### CODE REVIEW

**Comment by ale8k** [comment]
Please don't merge, will add issue templates soon, just wanted to get an exemplar and see if it is ok first.

**Comment by ijansky** [comment]
@alesstimec is there anything particular that should be among the checklist items for this repository?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Engineering checklist
+*Check only items that apply*
+
+- [ ] Documentation updated
+- [ ] Covered by unit tests
+- [ ] Covered by integration tests
+- [ ] Independent change*
+
+ _* For example, it does not require an sfproxy release before it can be merged._
```
Comment (alesstimec) [inline .github/PULL_REQUEST_TEMPLATE.md]: Please remove this comment: candid has nothing to do with sfproxy

File: .github/ISSUE_TEMPLATE/support.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Issue: Support
+
+## Ask your question clearly and concisely (Opportunity for detail is below)
```
Comment (alesstimec) [inline .github/ISSUE_TEMPLATE/support.md]: hmm.. not sure about the wording here.. seems like we don't trust our customers to provide a concise issue title..

File: .github/ISSUE_TEMPLATE/feature.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Issue: Feature
+> Note: The quickest way to see a new feature would be to submit a pull request, which will be reviewed when possible.
+## What feature?
```
Comment (alesstimec) [inline .github/ISSUE_TEMPLATE/feature.md]: s/What feature/Feature description/

File: .github/ISSUE_TEMPLATE/feature.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+<!--
+Please describe what functionality this feature pertains to.
+-->
+
+## Why include this feature?
```
Comment (alesstimec) [inline .github/ISSUE_TEMPLATE/feature.md]: I don't think we need this section. I believe users will describe the "why" in the feature description. We shouldn't ask them why they "feel" this feature should be included. Instead we should be interested in clear benefits of this feature.

File: .github/ISSUE_TEMPLATE/feature.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Issue: Feature
+> Note: The quickest way to see a new feature would be to submit a pull request, which will be reviewed when possible.
```
Comment (alesstimec) [inline .github/ISSUE_TEMPLATE/feature.md]: strictly speaking this is not absolutely true :) we have an architecture of candid in mind and outside users might not understand this fully. So a PR might lead to lengthy discussions an change requests. But i agree, we should not discourage anybody from proposing a PR. 

Instead, perhaps we should say something along the lines: "if you believe this feature requires a small change, feel free to propose a PR, which will be reviewed as soon as possible"

**Review by alesstimec — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/support.md (hunk 2 above)
Comment (ale8k) [reply .github/ISSUE_TEMPLATE/support.md]: That's fair, will leave it at 'Ask your question', I taken inspiration from a couple existing templates but I think you're right.

File: .github/ISSUE_TEMPLATE/feature.md (hunk 5 above)
Comment (ale8k) [reply .github/ISSUE_TEMPLATE/feature.md]: Ah ok, I'll update to your recommendation.

**Review by alesstimec — APPROVED** [review]

**Pull request merged**